projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -7214,8 +7214,17 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
                 const cells = this._dataRowList.map(x => x.cells.find(c => c.column === col));
                 cells.forEach((cell) => cellsContentWidths.push(cell?.nativeElement?.offsetWidth || 0));
                 const max = Math.max(...cellsContentWidths);
-                if (max === 0) {
-                    // cells not in DOM yet...
+                // in cases with template contains something, like a webcomponent,
+                // that renders fully only after it is already injected in the DOM,
+                // and initially renders as empty, skip measuring it.
+                let emptyCellWithPaddingOnly = 0;
+                if (cells.length > 0 && !!col.bodyTemplate) {
+                    const cellStyle = this.document.defaultView.getComputedStyle(cells[0].nativeElement);
+                    emptyCellWithPaddingOnly = parseFloat(cellStyle.paddingLeft) + parseFloat(cellStyle.paddingRight);
+                }
+
+                if (max === 0 || max <= emptyCellWithPaddingOnly) {
```

Comment:
Um, is it me or does this have the potential for looping hard and also possibly affecting the Angular product as well? Assuming there is a template that literally renders nothing on purpose for whatever reason (for example conditionally shown button), would this not get stuck trying to measure that empty column until something _is_ rendered?